tools/Config.mk

```diff
@@ -608,6 +620,42 @@ define DOWNLOAD
 	$(ECHO_END)
 endef
 
+# CLONE - Git clone repository. Initializes a new Git repository in the 
+#         folder on your local machine and populates it with the contents
+#         of the central repository.
+#         The third argument is an storage path. The second argument is used
+#         if it is not provided or is empty.
+# Example: $(call CLONE,$(URL_BASE),$(PATH),$(STORAGE_FOLDER))
+
+define CLONE
+	$(ECHO_BEGIN)"Clone: $(if $3,$3,$2) "
+	if [ -z $3 ]; then \
+		git clone --quiet $1 $2; \
+	else \
+		if [ ! -d $3 ]; then \
+			git clone --quiet $1 $3; \
+		fi; \
+		cp -fr $3 $2; \
+	fi
+	$(ECHO_END)
+endef
+
+# COMMITSHA - Check if the branche contain the named commit.
+#         Remove the folder if the commit is not present in the branch.
+#         The first argument is the repository folder on the local machine.
+#         The second argument is unique SHA-1 hash value.
+# Example: $(call COMMITSHA,$(GIT_FOLDER),$(COMMIT_SHA-1))
```

Comment:
I would call it `CHECK_COMMITSHA`. Also, I think the feature of this function should be different: instead of using `git branch --contains`, I think we should use `git rev-parse HEAD` to get the latest commit's SHA-1 and compare it with the provided one. Note: the branch may contain the commit but may be at a newer HEAD and, then, we wouldn't be using the same commit as expected. To solve that we should stick with the HEAD only.

Reply:
Forget about it: if it contains the commit, we would checkout to it soon. So, just some minor changes:
```suggestion
# CHECK_COMMITSHA - Check if the branch contains the commit SHA-1.
#         Remove the folder if the commit is not present in the branch.
#         The first argument is the repository folder on the local machine.
#         The second argument is a unique SHA-1 hash value.
# Example: $(call CHECK_COMMITSHA,$(GIT_FOLDER),$(COMMIT_SHA-1))
```

Reply:
Also, I think we can add more verbosity when the folder is non-existing and when it exists but contains no reference for the commit.